src/cljam/io/vcf/util.clj

```diff
@@ -233,3 +234,54 @@
           (update :info stringify-info)
           (update fmt-kw stringify-format)
           (merge (into {} (for [k sample-kws] [k (stringify-sample (v fmt-kw) (v k))])))))))
+
+(def ^:private ^:const long-breakend-regexp
+  ;;   pre-seq    [ or ]  chr    pos     [ or ]    post-seq
+  #"([ACGTN]*|\.)([\[\]])(.+?)(?::(\d+))([\[\]])([ACGTN]*|\.)")
+
+(def ^:private ^:const short-breakend-regexp
+  #"(\.?)([ATGCN]+)(\.?)")
+
+(defn parse-breakend
+  "Parses an ALT allele string of SVTYPE=BND.
+  Returns a map with mandatory keys:
+  - `:bases` bases that replaces the reference place
+  - `:join` `:before` or `:after`
+  and optional keys for a mate:
+  - `:chr` chromosome name of the mate sequence
+  - `:pos` genomic position of the mate sequence
+  - `:strand` strand of the mate sequence
```

Comment:
I'm not too confident, but I'm wondering if `strand` is a little confusing name here because it does not necessarily match the physical strand of the mate sequence (is that correct?).
It appears to me the VCF spec carefully uses the word `orientation` instead of `strand` for this meaning. What do you think of it?

Reply:
As you mentioned, `orientation` is clearly distinguished from `:strand` in the VCF spec.

> ```
> REF	ALT		Meaning
> s		t[p[	piece extending to the right of p is joined after t
> s		t]p]	reverse comp piece extending left of p is joined after t
> s		]p]t	piece extending to the left of p is joined before t
> s		[p[t	reverse comp piece extending right of p is joined before t
> ```

Summarizing the above:

|ALT|`:strand`|orientation|`:join`|
|:-:|:-:|:-:|:-:|
|`t[p[`|`:forward`|right|`:after`|
|`t]p]`|`:reverse`|left|`:after`|
|`]p]t`|`:forward`|left|`:before`|
|`[p[t`|`:reverse`|right|`:before`|

I chose to omit the `:orientation` since one of the three flags is redundant for four possible modes.

Reply:
Thank you for elaborating on the difference between strand and orientation! That does make much sense.